### PR TITLE
[#158333555] Stop double emitting metrics

### PR DESCRIPTION
**Pull request by richardTowers**
## What

We want to only emit metrics from a single instance of this collector. We have done this using the Locket client and server, as done by Cloud Foundry components such as TPS.

## How to review

* code review
* run `make unit`
* check integration tests passed:
  https://concourse.build.ci.cloudpipeline.digital/teams/main/pipelines/rds-metric-collector
  Note: the integration tests have been flakey. A separate story has been raised [#159328199](https://www.pivotaltracker.com/story/show/159328199).

Optionally, you can deploy it with https://github.com/alphagov/paas-cf/pull/1462 and go poke at the logs.

## Who

Anyone but me or @richardTowers

### CODE REVIEW

File: ci/blackbox/rds_metric_collector_suite_test.go [hunk 1]
```diff
@@ -37,14 +42,21 @@ var (
...
 
+		// Compile test Locket server
+		mockLocketServer := testhelpers.MockLocketServer{}
+		mockLocketServer.Build()
+		mockLocketServerSession = mockLocketServer.Run(fixturesPath, "alwaysGrantLock")
+		Eventually(mockLocketServerSession.Buffer).Should(gbytes.Say("grpc.grpc-server.started"))
+
 		// Update config
-		rdsBrokerConfig, err = rdsconfig.LoadConfig("../../fixtures/broker_config.json")
+		rdsBrokerConfig, err = rdsconfig.LoadConfig(fixturesPath + "/broker_config.json")
```
Comment (blairboy362) [inline ci/blackbox/rds_metric_collector_suite_test.go]: https://golang.org/pkg/path/#Join

File: ci/blackbox/rds_metric_collector_suite_test.go [hunk 2]
```diff
@@ -72,9 +84,9 @@ func TestSuite(t *testing.T) {
 
 		// Start a fake server for loggregator
 		fakeLoggregator, err = helpers.NewFakeLoggregatorIngressServer(
-			"../../fixtures/server.crt",
-			"../../fixtures/server.key",
-			"../../fixtures/CA.crt")
+			fixturesPath + "/loggregator-server.cert.pem",
```
Comment (blairboy362) [inline ci/blackbox/rds_metric_collector_suite_test.go]: https://golang.org/pkg/path/#Join

File: scripts/generate-cert-fixtures.sh [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  rm -f fixtures/client.csr.pem
+  rm -f fixtures/locket-server.csr.pem
+  rm -f fixtures/loggregator-server.csr.pem
+  rm -f index.txt index.txt.attr index.txt.old index.txt.attr.old
+  rm -f serial serial.old
+}
+trap 'cleanup' EXIT
+
+# Create CA certificate
+openssl genrsa -out fixtures/ca.key.pem 2048
```
Comment (blairboy362) [inline scripts/generate-cert-fixtures.sh]: Thanks for automating this stuff.

File: pkg/scheduler/scheduler.go [hunk 4]
```diff
@@ -153,14 +153,24 @@ func (w *Scheduler) Start() error {
 			}
 		}
 	})
-	return err
+	if err != nil {
+		w.logger.Debug("error-starting-scheduler")
+		return err
+	}
+	close(ready)
```
[the remark this answers is not in the document]
Comment (henrytk) [reply pkg/scheduler/scheduler.go]: In this case, yes. This `ready` channel is passed to the process runner for this purpose specifically[1]. It's a way of signalling to Ifrit, the process manager, that this subprocess has been set up correctly and can start doing its thing.

[1] https://godoc.org/github.com/tedsuo/ifrit#Runner

File: scripts/generate-cert-fixtures.sh (hunk 3 above)
Comment (henrytk) [reply scripts/generate-cert-fixtures.sh]: It's what I do ;)

File: pkg/scheduler/scheduler.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (blairboy362) [reply pkg/scheduler/scheduler.go]: Awesome. Thanks for the link @henrytk

**Pull request merged**